src/main/java/org/skife/jdbi/v2/PreparedBatch.java

```diff
@@ -109,7 +110,22 @@ public PreparedBatch define(final Map<String, ? extends Object> values)
                         getContext(),
                         getContainerMapperRegistry());
             }
-        });
+        }, null);
+
+    }
+
+    @SuppressWarnings("unchecked")
+    public <GeneratedKeyType> GeneratedKeys<GeneratedKeyType> executeAndGenerateKeys(final ResultSetMapper<GeneratedKeyType> mapper,
+                                                                                     String[] columnNames) {
```

Comment:
This could be `String...` and then varargs invocation would allow you to elide the `new String[] { ... }` dance, although maybe being explicit is better here.